### PR TITLE
2024-06 Update

### PR DESCRIPTION
**Pull request by viragpurnam**
Used tycho version 4.0.4 instead of 4.0.8 (Latest).
Maven version to be used 3.9.8. There is no need to upgrade to Java 21. 
@thegreystone : Could you please review the code changes.

### CODE REVIEW

**Comment by thegreystone** [comment]
Even though 17 works, shouldn't we switch to 21? 

https://www.eclipse.org/setups/sponsor/?scope=Eclipse%20IDE&campaign=2024-06

Looks risky to use 17, given the wording there:
"The June 12th release of the Eclipse IDE 2024-06 - 4.32 requires Java 21."

**Comment by viragpurnam** [comment]
Ok.. I will update my PR with JAVA 21

**Comment by viragpurnam** [comment]
Hi @thegreystone, I have done a new commit. Could you please review that. I have commented the goal "build-qualifier". Not very much sure if that is needed.

File: releng/platform-definitions/platform-definition-2024-06/platform-definition-2024-06.target [hunk 1]
```diff
@@ -91,5 +92,5 @@
             <repository location="https://archive.eclipse.org/technology/babel/update-site/R0.20.0/2022-12/"/>
         </location>
     </locations>
-    <targetJRE path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-21"/>
+    <targetJRE path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17"/>
```
Comment (thegreystone) [inline releng/platform-definitions/platform-definition-2024-06/platform-definition-2024-06.target]: Shouldn't this stay 21?

Comment (viragpurnam) [reply releng/platform-definitions/platform-definition-2024-06/platform-definition-2024-06.target]: Yes, I missed that. I have done the changes and committed the code.

File: pom.xml [hunk 2]
```diff
@@ -256,9 +256,9 @@
 					<execution>
 						<id>build-qualifier</id>
 						<phase>validate</phase>
-						<goals>
+						<!--<goals>
```
Comment (viragpurnam) [inline pom.xml:259]: Hi @thegreystone, This is the main change which is making build success. I am not sure on the impact. This generates the build qualifier according to the [rules described in the PDE documentation](https://help.eclipse.org/latest/index.jsp?topic=%2Forg.eclipse.pde.doc.user%2Ftasks%2Fpde_version_qualifiers.htm). 
I just compared the build with and without this goal. There is no difference in the target.

**Review by thegreystone — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (thegreystone) [reply pom.xml:259]: Did you try to make a release build with these changes? Could you try building a release build and check what the about box says? It should look something along the lines of:

Azul Mission Control 9.0.0 CA
(Build id: 15, Qualifier: 202405211355, Change id: f58490a6bddde45d8862aab25795ebdb3b1b4d08)

Comment (viragpurnam) [reply pom.xml:259]: Sure. I will check this.

**Pull request merged**